### PR TITLE
Address #1996

### PR DESCRIPTION
**Pull request by ben-arnao**
Let users know that weight decay optimizers are not compatible with lookahead

### CODE REVIEW

**Comment by ben-arnao** [comment]
@WindQAQ I have uploaded a test. I do apologize if it is not good, i'm not too familiar with creating tests.

**Comment by WindQAQ** [comment]
LGTM! We can use `pytest.raises` to catch the exception, such as
https://github.com/tensorflow/addons/blob/993f2d8587d20fc319dde4e2eb46b0f565d42fb7/tensorflow_addons/utils/tests/keras_utils_test.py#L29-L30

**Comment by WindQAQ** [comment]
Just figure out than one common use is to wrap RectifiedAdam with Lookahead, and it seems to work probably.

**Comment by ben-arnao** [comment]
Yea, actually sees like RAdam has weight_decay... which causes the tests to fail

```
from tensorflow_addons.optimizers import RectifiedAdam
from tensorflow.keras.optimizers import Adam

print(hasattr(Adam(), "weight_decay"))
print(hasattr(RectifiedAdam(), "weight_decay"))
```

Not sure if RAdam should have weight decay... the vanilla Adam does not.

**Comment by WindQAQ** [comment]
The implementation from author also includes weight decay.

https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam/radam.py

**Comment by seanpmorgan** [comment]
See the colab in the #1996:
https://colab.research.google.com/drive/1b-8JSEMx0Ruqfue9L092sCQxq7-KJaZW?usp=sharing
